### DISCUSSION TITLE
Unable to Find Time or Velocity Configuration for Generating Position Trajectory

### DISCUSSION THREAD

**Discussion by roboticszeng** [NVlabs/curobo#147] [Software Q&A] [answered]
### Description
I am trying to generate a position trajectory in Curobo with the following requirements:

- Total motion time: 10 seconds
- Joint motion maximum speed: 0.5 rad/s

However, I couldn't find suitable configuration options in the `load_robot_config`. I need assistance in configuring Curobo to meet these specifications.

### Details
- **Total Motion Time:** 10 seconds
- **Joint Maximum Speed:** 0.5 rad/s

### Steps to Reproduce
[Provide any relevant code snippets or steps you've taken so far.]

### Expected Behavior
[Explain what you expected to happen.]

### Actual Behavior
[Describe the actual behavior or error messages you encountered.]

### Additional Information
[Provide any additional context or information that might be helpful for addressing the issue.]

**Comment by roboticszeng** [comment]
I have attempted to adjust the max_acceleration and max_jerk parameters in the YAML file, as well as the scaling factors for velocity, acceleration, and jerk in the load_robot_config section. While I observed some improvement when setting these values relatively small, the trajectory planning fails whenever I make them too small, resulting in a 'TrajOpt Failed' error. Could you please provide guidance on how to appropriately configure these parameters

**Comment by balakumar-s** [comment]
To plan for small velocity ranges:

Try a velocity_scale of 0.5 or lower here: https://github.com/NVlabs/curobo/blob/c09d94908d8ad52f557ed59195c1ceb2d0434d65/src/curobo/wrap/reacher/motion_gen.py#L198

**Reply by roboticszeng** [reply]
Dear balakumar-s,


Thank you for your response. I've noticed that increasing the `<velocity>` values in the URDF or adjusting the `velocity_scale` seems to improve the success rate of the planning. However, I'm still uncertain about the underlying reasons. My specific scenario involves a 7-dof robotic arm without considering environmental collisions or self-collisions. The goal is to plan trajectories between two sets of randomly generated joint angles.

Traditionally, with methods like trapezoidal planning or fifth-degree polynomials, the theoretical success rate is expected to be 100%. Nevertheless, I'm puzzled by the fact that when using Curobo for planning, the success rate is considerably lower, especially when the velocity is set to very low values (less than 0.2rad/s). I'm wondering if there are specific parameters I might be overlooking or if there are any adjustments needed.

Any insights or guidance you can provide on which parameters might be affecting the planning success, especially at low velocities, would be greatly appreciated.

Thank you!

**Reply by balakumar-s** [reply, marked as answer]
1. If you don't change the velocity scaling, does success rate of cuRobo reach 100%? 

2. Can you add this parameter `maximum_trajectory_dt=2.0` to here https://github.com/NVlabs/curobo/blob/c09d94908d8ad52f557ed59195c1ceb2d0434d65/src/curobo/wrap/reacher/motion_gen.py#L198
and see if that at least gives more success rate? 

We tuned the weights for smoothing for speeds in the range of 1-2 radians per second. These weights worked down to 0.5 radians per second. Anything lower might need some re-tuning. 

I think this will be a good unit test for us and we will try to reproduce this issue on an existing robot in our repository.

**Reply by balakumar-s** [reply]
We have added a new set of weights for slow speeds and also added a unit test: https://github.com/NVlabs/curobo/blob/main/tests/motion_gen_speed_test.py

Let us know if these help.

**Comment by roboticszeng** [comment]
> We have added a new set of weights for slow speeds and also added a unit test: https://github.com/NVlabs/curobo/blob/main/tests/motion_gen_speed_test.py
> 
> Let us know if these help.

Thank you! Now it's running smoothly. I suspect there were some internal parameters that weren't tuned properly.
